File: 백준/Bronze/2798. 블랙잭/블랙잭.c
#include<stdio.h>
int main(void){
    int n,m;
    scanf("%d%d",&n,&m);
    int sum, temp = 0;
    int arr[100] = {0,};
    
    for(int i = 0; i < n; i++){
        scanf("%d",&arr[i]);
    }
    for(int i = 0; i < n; i++){
        for(int j = i + 1; j < n; j++){
            for(int k = j + 1; k < n; k++){
                sum = arr[i] + arr[j] + arr[k];
                if(temp < sum && sum <= m)
                    temp = sum;
            }
        }
    }
    printf("%d\n",temp);
    return 0;
}

File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n,m = map(int,input().split())
arr = list(map(int,input().split()))
sum = 0
for i in range(n):
    for j in range(n):
        for k in range(n):
            if arr[i] == arr[j] or arr[i] == arr[k] or arr[j] == arr[k]:
                continue
            else:
                if arr[i] + arr[j] + arr[k] <= m and sum < arr[i] + arr[j] + arr[k]:
                   sum = arr[i] + arr[j] + arr[k]
                elif arr[i] + arr[j] + arr[k] > m:
                    continue
print(sum)